### PR TITLE
AwesomeBar(): Add onScroll parameter to be invoked at the beginning of the user performing a scroll gesture.

### PR DESCRIPTION
**Pull request by pocmo**
📜

### CODE REVIEW

File: samples/compose-browser/src/main/java/org/mozilla/samples/compose/browser/browser/BrowserScreen.kt [hunk 1]
```diff
@@ -206,6 +209,8 @@ private fun Suggestions(
         )
     }
 
+    val keyboardController = LocalSoftwareKeyboardController.current
+
```
Comment (mcarare) [inline samples/compose-browser/src/main/java/org/mozilla/samples/compose/browser/browser/BrowserScreen.kt:213]: It's nice this thing exists now 😀

File: components/compose/awesomebar/src/main/java/mozilla/components/compose/browser/awesomebar/AwesomeBar.kt [hunk 2]
```diff
@@ -52,7 +54,8 @@ fun AwesomeBar(
             colors,
             orientation,
             onSuggestionClicked,
-            onAutoComplete
+            onAutoComplete,
```
[the remark this answers is not in the document]
Comment (pocmo) [reply components/compose/awesomebar/src/main/java/mozilla/components/compose/browser/awesomebar/AwesomeBar.kt:57]: Would that make things more clear? I try to name parameters if the value doesn't give any hints (for example if it's just `true`). But that didn't seem necessary here.

**Review by mcarare — APPROVED** [review]
LGTM

**Comment by mcarare** [comment]
Maybe this would be nice to land before https://github.com/mozilla-mobile/fenix/pull/21191. Either way is fine since it will be only in Nightly, I guess.

File: components/compose/awesomebar/src/main/java/mozilla/components/compose/browser/awesomebar/AwesomeBar.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (MozillaNoah) [reply components/compose/awesomebar/src/main/java/mozilla/components/compose/browser/awesomebar/AwesomeBar.kt:57]: I guess I just wasn't sure in the world of Compose if EVERYTHING should be spelled-out or just on the composeables where there's a ton of optional parameters.

**Comment by pocmo** [comment]
> Maybe this would be nice to land before [mozilla-mobile/fenix#21191](https://github.com/mozilla-mobile/fenix/pull/21191). Either way is fine since it will be only in Nightly, I guess.

Agree, I will land this here first and then wait for it to arrive in Fenix before merging the other PR.

File: samples/compose-browser/src/main/java/org/mozilla/samples/compose/browser/browser/BrowserScreen.kt (hunk 1 above)
Comment (pocmo) [reply samples/compose-browser/src/main/java/org/mozilla/samples/compose/browser/browser/BrowserScreen.kt:213]: Interesting. 🤔 

I guess keeping focus is fine and preferred in situations with hardware keyboards, where we wouldn't want to remove focus and allow them to continue typing, even if they scroll the list? I haven't tried that though.

Comment (mcarare) [reply samples/compose-browser/src/main/java/org/mozilla/samples/compose/browser/browser/BrowserScreen.kt:213]: For this particular case, I guess it makes no difference, you can still use the complete arrow in both cases. Just wanted to keep it in mind for similar use cases where one solution might be better to integrate than the other.

**Pull request merged**